src/udiskslinuxencrypted.c

```diff
@@ -803,10 +804,143 @@ handle_change_passphrase (UDisksEncrypted        *encrypted,
+                                             "Device %s does not appear to be a LUKS device",
+                                             udisks_block_get_device (block));
+      goto out;
+    }
+
+  error = NULL;
+  if (!udisks_daemon_util_get_caller_uid_sync (daemon, invocation, NULL /* GCancellable */, &caller_uid, NULL, NULL, &error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation, error);
+      g_clear_error (&error);
+      goto out;
+    }
+
+  /* Fail if device is not unlocked */
+  cleartext_object = udisks_daemon_wait_for_object_sync (daemon,
+                                                         wait_for_cleartext_object,
+                                                         g_strdup (g_dbus_object_get_object_path (G_DBUS_OBJECT (object))),
+                                                         g_free,
+                                                         0, /* timeout_seconds */
+                                                         NULL); /* error */
+  if (cleartext_object == NULL)
+    {
+      g_dbus_method_invocation_return_error (invocation,
+                                             UDISKS_ERROR,
+                                             UDISKS_ERROR_FAILED,
+                                             "Device %s is not unlocked",
+                                             udisks_block_get_device (block));
+      goto out;
+    }
+  cleartext_block = udisks_object_peek_block (cleartext_object);
+
+  action_id = "org.freedesktop.udisks2.modify-device";
+  /* Translators: Shown in authentication dialog when the user
+   * requests resizing a encrypted block device.
+   *
+   * Do not translate $(drive), it's a placeholder and
+   * will be replaced by the name of the drive/device in question
+   */
+  message = N_("Authentication is required to resize the encrypted device $(drive)");
+  if (! udisks_daemon_util_setup_by_user (daemon, object, caller_uid))
+    {
+      if (udisks_block_get_hint_system (block))
+        {
+          action_id = "org.freedesktop.udisks2.modify-device-system";
+        }
+      else if (! udisks_daemon_util_on_user_seat (daemon, UDISKS_OBJECT (object), caller_uid))
+        {
+          action_id = "org.freedesktop.udisks2.modify-device-other-seat";
+        }
+    }
+
+  /* Check that the user is actually authorized to resize the device. */
+  if (! udisks_daemon_util_check_authorization_sync (daemon,
+                                                     object,
+                                                     action_id,
+                                                     options,
+                                                     message,
+                                                     invocation))
+    goto out;
+
+  job = udisks_daemon_launch_simple_job (daemon,
+                                         UDISKS_OBJECT (object),
+                                         "encrypted-resize",
```

Comment:
Please add this job type to `udisks/udisksclient.c` (function `udisks_client_get_job_description`, line 2476) and `data/org.freedesktop.UDisks2.xml` (`org.freedesktop.UDisks2.Job` interface description, line 2455).

Also please add the new generated functions to `doc/udisks2-sections.txt.in.in` (`UDisksEncrypted` section).

Reply:
Done.

Reply:
I noticed this:

```
diff --git a/doc/udisks2-sections.txt.in.in b/doc/udisks2-sections.txt.in.in
index 772709fc..3aa7a1f0 100644
--- a/doc/udisks2-sections.txt.in.in
+++ b/doc/udisks2-sections.txt.in.in
@@ -912,7 +912,7 @@ udisks_filesystem_complete_unmount
 udisks_filesystem_call_resize
 udisks_filesystem_call_resize_finish
 udisks_filesystem_call_resize_sync
-disks_filesystem_complete_resize
+udisks_filesystem_complete_resize
 udisks_filesystem_call_check
 udisks_filesystem_call_check_finish
 udisks_filesystem_call_check_sync
```